x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/settings/AbstractSettingsService.java

```diff
@@ -22,27 +18,4 @@ public AbstractSettingsService(final Client client) {
         this.client = client;
     }
 
-    @Override
-    public S getSettings(String settingsId) throws SettingsNotFoundException, InvalidSettingsException {
```

Comment:
getSettings depend on the service type, as CurationsService need to do a query instead of retrieval by ID.